benchmarks/benchmarks.rb

```diff
@@ -74,22 +74,24 @@ def keyword_args(a:, b:)
   CLASS
 end
 
+# We pre-create argument lists for our memoized methods with arguments, so that
+# our benchmarks are running the exact same inputs for each case.
+N_UNIQUE_ARGUMENTS = 100
+ARGUMENTS = Array.new(N_UNIQUE_ARGUMENTS) { |i| [i, i + 1] }
+
 # We benchmark different cases separately, to ensure that slow performance in
 # one method or code path isn't hidden by fast performance in another.
-N_UNIQUE_ARGUMENTS = 1_000
-N_CALLS_PER_UNIQUE_ARGUMENTS = 1_000
-N_INTERATIONS_WITHOUT_ARGUMENTS = N_UNIQUE_ARGUMENTS *
-                                  N_CALLS_PER_UNIQUE_ARGUMENTS
-ARGUMENTS = Array.new(N_UNIQUE_ARGUMENTS) { |i| [i, i + 1] }
 
 Benchmark.ips do |x|
   x.config(suite: suite)
   BENCHMARK_GEMS.each do |benchmark_gem|
-    example_class = Object.const_get("#{benchmark_gem.klass}Example")
-    x.report("#{benchmark_gem.klass} (#{benchmark_gem.version}): no_args") do
-      instance = example_class.new
-      N_INTERATIONS_WITHOUT_ARGUMENTS.times { instance.no_args }
-    end
+    instance = Object.const_get("#{benchmark_gem.klass}Example").new
+
+    # Run once to memoize the result value, so our benchmark only tests memoized
+    # retrieval time.
+    instance.no_args
+
+    x.report("#{benchmark_gem.benchmark_name}: no_args") { instance.no_args }
```

Comment:
I mentioned this in my original benchmarks PR, but I'm not sure what to do about the fact that the `benchmark-ips` gem has a vague recommendation that very fast blocks might be better off using the `do |times|` construction: https://github.com/evanphx/benchmark-ips#synopsis

I couldn't find a lot of details on that, but my guess is that the overhead it's referring to is block execution overhead and that using a while loop might be lighter-weight. It seems a little hacky and makes me uneasy though, and I'm pretty sure more recent Ruby versions have optimized block overhead more so I thought it made sense to just use the simpler block version.

Reply:
I think it's fine to start with this, as it is showing clear distinctions already -- we could always experiment with changing it later

Reply:
(comment on the PR) I think this is fine, as it's already revealing significant performance differences -- we can always try out the other form later and see if you like it then